cubic-server/CompressionUtils.cpp

```diff
@@ -0,0 +1,90 @@
+#include <cstddef>
+#include <cstdint>
+#include <unistd.h>
+#include <vector>
+#include <zlib.h>
+
+#include "Server.hpp"
+
+static void add_buffer_to_vector(std::vector<uint8_t> &vector, const uint8_t *buffer, size_t length)
+{
+    for (size_t character_index = 0; character_index < length; character_index++) {
+        char current_character = buffer[character_index];
+        vector.push_back(current_character);
+    }
+}
+
+int compressVector(const std::vector<uint8_t> &source, std::vector<uint8_t> &destination)
+{
+    unsigned long source_length = source.size();
+    size_t destination_length = compressBound(source_length);
+
+    uint8_t *destination_data = (uint8_t *) malloc(destination_length);
```

Comment:
What about no ?

Reply:
Yes, I will use that, cause zlib is a C library, I don't really have a choice :(